### ISSUE TITLE
enhance.equalize_clahe: an image as large as the grid raises a raw padding error

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4783]
`kornia.enhance.equalize_clahe` (and `RandomClahe`, which delegates to it) raises a named `ValueError` when the grid is larger than the image, but an image exactly as large as the grid — the default `grid_size=(8, 8)` on an `8 x 8` image — falls through that check and fails inside the reflect padding with a raw torch error.

## Repro

```python
import torch, kornia
from kornia.augmentation import RandomClahe

for size in (7, 8, 9):
    try:
        kornia.enhance.equalize_clahe(torch.rand(1, 1, size, size), 40.0, (8, 8))
        print(size, "ok")
    except Exception as e:
        print(size, type(e).__name__, e)
try:
    RandomClahe(p=1.0)(torch.rand(1, 1, 8, 8))
except Exception as e:
    print("RandomClahe", type(e).__name__, e)
```

## Observed

```
7 ValueError Cannot compute tiles on the image according to the given grid size
8 RuntimeError Argument #4: Padding size should be less than the corresponding input dimension, but got: padding (0, 8) at dimension 3 of input [1, 1, 8, 8]
9 ok
RandomClahe RuntimeError Argument #4: Padding size should be less than the corresponding input dimension, but got: padding (0, 8) at dimension 3 of input [1, 1, 8, 8]
```

## Expected

The tile size is `ceil(8 / 8) = 1`, which `equalize_clahe` rounds up to an even `2`, so the image is reflect-padded to `16 x 16` — a pad of `8` on an axis of length `8`, which reflect padding cannot do. Either handle the case (a tile of one pixel, or a padding mode that can extend that far), or extend the existing `ValueError` so it names the smallest image the grid admits.

## Why it is a bug

The function has a named size check, but its bound is off: it lets through an input that its own padding then rejects with torch's message.

## Environment

kornia `main` @ 57cd68e6f, torch 2.14.0, python 3.11, macOS arm64 cpu.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
